.github/workflows/integration.yml

```diff
@@ -0,0 +1,197 @@
+          key: ${{ runner.os }}-cpp-${{ github.sha }}
+
+  go-unix:
+    name: "Go/${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ["macos-latest", "ubuntu-latest"]
+    env:
+      CGO_ENABLED: "1"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.6
+          check-latest: true
+          cache: true
+          cache-dependency-path: go/adbc/go.sum
+      - name: Build shared lib
+        id: build-shared
+        shell: bash -l {0}
+        run: |
+          export PATH=$RUNNER_TOOL_CACHE/go/1.18.6/x64/bin:$PATH
+          pushd go/adbc/pkg
+          make
+          mkdir -p $HOME/local/lib
+          mv lib* $HOME/local/lib/          
+          echo "CACHEPATH=$HOME/local" >> $GITHUB_OUTPUT
+          popd
+      - name: Cache builds
+        uses: actions/cache/save@v3
+        with:
+          path: ${{ steps.build-shared.outputs.CACHEPATH }}
+          key: ${{ runner.os }}-go-${{ github.sha }}
+
+  run-tests:
+    name: "Integration/${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix: # macos-latest having issues with docker?
+        os: ["ubuntu-latest"]
+    needs: [cpp-conda-unix, go-unix]
+    services:
+      dremio:
+        image: dremio/dremio-oss:latest
+        env:
+          DREMIO_JAVA_EXTRA_OPTS: "-Ddebug.addDefaultUser=true -Ddremio.eula.disabled=true"
+        options: >-
+          --health-cmd "curl --fail http://localhost:9047/ || exit 1"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 9047:9047
+          - 32010:32010
+          - 31010:31010
+          - 45678:45678
+    steps:
+      - name: Get CachePath
+        id: cache-path
+        shell: bash -l {0}
+        run: |
+          echo "CACHEPATH=$HOME/local" >> $GITHUB_OUTPUT
+      - name: Restore CPP build
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ steps.cache-path.outputs.CACHEPATH }}
+          key: ${{ runner.os }}-cpp-${{ github.sha }}
+      - name: Restore Go build
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ steps.cache-path.outputs.CACHEPATH }}
+          key: ${{ runner.os }}-go-${{ github.sha }}
+      - name: Prove it worked
+        shell: bash -l {0}
+        run: |
+          ls -l $HOME/local/**
+
+      - name: Bootstrap Dremio instance
+        shell: bash -l {0}
+        id: bootstrap
+        run: |          
+          LOGIN=`curl -X POST -H "Content-Type: application/json" "http://localhost:9047/apiv2/login" \
+            -d '{"userName":"dremio","password":"dremio123"}'`
+          echo "LOGIN_RESP=$LOGIN" >> $GITHUB_OUTPUT
+
+      - name: Print Token
+        shell: bash -l {0}
+        run: echo "_dremio${{ fromJSON(steps.bootstrap.outputs.LOGIN_RESP).token }}"
```

Comment:
Seems the actual test step isn't implemented yet?

Reply:
yea, not yet. that's next. All of this so far was me making sure i get the setup and config working properly.